spec/views/results/no_results_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+describe 'results/no_results.html.erb', type: :view do
+  context 'Scotland' do
+    let(:html) do
+      render partial: 'results/no_results', locals: { country: 'Scotland', devolved_nation: true }
+    end
+
+    it 'renders Scottish teacher training website' do
+      expect(html).to match('This service is for courses in England')
+      expect(html).to have_link('Learn more about teacher training in Scotland', href: 'https://teachinscotland.scot/')
+    end
+  end
+
+  context 'Wales' do
+    let(:html) do
+      render partial: 'results/no_results', locals: { country: 'Wales', devolved_nation: true }
+    end
+
+    it 'renders Welsh teacher training website' do
+      expect(html).to match('This service is for courses in England')
+      expect(html).to have_link('Learn more about teacher training in Wales',
+                                href: 'https://www.discoverteaching.wales/routes-into-teaching/')
+    end
+  end
+
+  context 'Northern Ireland' do
+    let(:html) do
+      render partial: 'results/no_results', locals: { country: 'Northern Ireland', devolved_nation: true }
+    end
+
+    it 'renders Northern Irish teacher training website' do
+      expect(html).to match('This service is for courses in England')
+      expect(html).to have_link('Learn more about teacher training in Northern Ireland',
+                                href: 'https://www.education-ni.gov.uk/articles/initial-teacher-education-courses-northern-ireland')
+    end
+  end
+
+  context 'England' do
+    let(:html) { render partial: 'results/no_results', locals: { country: 'England', devolved_nation: false } }
+    let(:results_view) { described_class.new(query_parameters: { 'funding' => '8' }) }
+
+    it 'renders try another search text' do
+      expect(html).to match('There are no courses matching your search')
```

Comment:
This test currently fails because `@results_view` is `nil`. I thought the `let` on line 41 would take care of this, but I must be missing something.

Reply:
I've now sorted this too.